docs/spec/spec.md

```diff
@@ -197,52 +259,158 @@ channel._
+destination information. This is typically stored in the object's `status`, as
+this information may be generated by the controller. More details in
+[`Addressable` interface](interfaces.md#addressable).
+
+| Field Name | Field Type              | Description                                 | Constraints     |
+| ---------- | ----------------------- | ------------------------------------------- | --------------- |
+| `url`      | [`apis.URL`](#apis.url) | The URI name of the endpoint for the reply. | Must be an URL. |
+
+### apis.URL
 
-| Field      | Type         | Description                                         | Constraints |
-| ---------- | ------------ | --------------------------------------------------- | ----------- |
-| kind       | String       | The backing channel CRD                             |             |
-| apiVersion | String       | API version of backing channel CRD                  |             |
-| spec       | RawExtension | Spec to be passed to backing channel implementation |             |
+`URL` is an alias of [`url.URL`](https://golang.org/pkg/net/url/#URL) . It has
+custom JSON marshal methods that enable it to be used in Kubernetes CRDs such
+that the CRD resource will have the URL, but operator code can work with the
+`url.URL` struct.
 
-### Destination
+### duckv1.KReference
 
-| Field           | Type            | Description                                                                                          | Constraints                   |
-| --------------- | --------------- | ---------------------------------------------------------------------------------------------------- | ----------------------------- |
-| ref<sup>1</sup> | ObjectReference | The Subscription UID this SubscriberSpec was resolved from.                                          | Must adhere to _Addressable_. |
-| uri<sup>1</sup> | apis.URL        | Either an absolute URL (if ref is not specified). Resolved using the base URI from ref if specified. | Must be a URL.                |
+`KReference` contains enough information to refer to another object.
+
+| Field Name   | Field Type | Requirement | Description                                                                                                              | Default Value                      | Constraints |
+| ------------ | ---------- | ----------- | ------------------------------------------------------------------------------------------------------------------------ | ---------------------------------- | ----------- |
+| `kind`       | `string`   | Required    | [Kind of the referent.](https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds) |                                    |             |
+| `namespace`  | `string`   | Optional    | [Namespace of the referent.](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/)              | defaulted to the object holding it |             |
```

Comment:
nit for the "defaulted to the object holding it", I wonder if a word like embedding might be clearer.

Reply:
Updated.